src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,153 @@
+  constructor(datasource) {
+    super(datasource);
+    this.meetingsObservables = {};
+  }
+
+  /**
+   * Register the SDK meeting plugin to the device
+   * and sync the meeting collection from the server.
+   */
+  async connect() {
+    await this.datasource.meetings.register();
+    await this.datasource.meetings.syncMeetings();
+  }
+
+  /**
+   * Unregister the sdk meeting plugin from the device.
+   */
+  async disconnect() {
+    await this.datasource.meetings.unregister();
+  }
+
+  /**
+   * Retrieve the local device media and add them with
+   * the given default media settings to the meeting.
+   *
+   * @param {string} meetingID  meeting ID to retrieve the SDK meeting object to add local media to
+   */
+  async addLocalMedia(meetingID) {
+    const meeting = this.datasource.meetings.getMeetingByType('id', meetingID);
+
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+    const [localStream, localShare] = await meeting.getMediaStreams(mediaSettings);
+
+    return meeting.addMedia({
+      localShare,
+      localStream,
+      mediaSettings,
+    });
+  }
+
+  /**
+   * Returns a meeting object with media attached based on a given event type.
+   *
+   * @param {Meeting} meeting outdated meeting object
+   * @param {Object} media   new media stream to attach to the meeting object based on a given event type
+   * @returns {Object}
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  attachMedia(meeting, {type, stream}) {
+    let newMeeting;
+
+    switch (type) {
+      case LOCAL:
+        newMeeting = {...meeting, localAudio: stream.getAudioTracks()[0], localVideo: stream.getVideoTracks()[0]};
+        break;
+      default:
+        newMeeting = {...meeting};
+        break;
+    }
+
+    return newMeeting;
+  }
+
+  /**
+   * Returns a meeting custom shaped meeting object retrieved from the collection.
+   *
+   * @param {string} ID ID of the meeting to fetch
+   * @returns {Meeting}
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  fetchMeeting(ID) {
+    const meeting = this.datasource.meetings.getMeetingByType('id', ID);
+
+    return meeting
+      ? {
+          ID: meeting.id,
+          title: meeting.sipuri,
+          localVideo: null,
+          localAudio: null,
+          localShare: null,
+          remoteStream: null,
+          remoteShare: null,
+        }
+      : undefined;
+  }
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID ID of meeting to get
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  getMeeting(ID) {
+    if (!(ID in this.meetingsObservables)) {
+      const meeting = this.fetchMeeting(ID);
+      const getMeeting$ = Observable.create((observer) => {
+        if (meeting) {
```

Comment:
Let's move the `fetchMeeting` to inside this observable since the meeting object isn't used anywhere outside this scope:
```js
const getMeeting$ = Observable.create((observer) => {
  const meeting = this.fetchMeeting(ID);
  
  if (meeting) {
```